src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -63,22 +63,21 @@ export default abstract class QueryBase implements QueryInterface {
     deposit: Deposit,
     amountToRelay: BigNumberish,
     relayAddress = DEFAULT_SIMULATED_RELAYER_ADDRESS
-  ): Promise<BigNumberish> {
+  ): Promise<{ gasCost: BigNumberish; gasTokenCost: BigNumberish }> {
```

Comment:
fwiw I'm not in love with the naming `gasCost` and `gasTokenCost`. I'm wondering whether `gasCostNative` and `gasCostToken` might be a better division? We do already have a concept of the "native" gas cost in the relayer, but that can be adjusted if it makes the e2e naming more cohesive.

Reply:
I think I'd prefer it to be something like `nativeGasCost` and `tokenGasCost`. There's a few ways to work this, but I think it would be a cleaner naming scheme not to leave one value as `gasCost`

Reply:
I settled on nativeGasCost and tokenGasCost in the end :+1: